department-of-veterans-affairs/caseflow
# Remove judge name from case details hearing section

Resolves [#1757: Hide Judge Name in Hearings Section](https://vajira.max.gov/browse/CASEFLOW-1757).

### Description
We don't want VSOs/POAs to be able to see the judge who is assigned to a hearing, because we don't want them to have access to this information before a hearing.  I shielded this information in the Case Details Page's Hearings Section.  Please see attached screenshots. 

### Acceptance Criteria
- [ ] VSO user cannot see judge name on case details
- [ ] Non-vso user can see judge name on case details page

### Testing Plan
1. Log in as `BILLIE_VSO`

2. Create a veteran, appeal, and associated hearing in the rails console.
```
RequestStore[:current_user] = User.system_user
veteran = FactoryBot.create(:veteran, first_name: "Chibueze", last_name: "Vanscoy", file_number: 800_888_001)
appeal = FactoryBot.create(:appeal, :hearing_docket, closest_regional_office: "RO39", veteran_file_number: veteran.file_number)
hearing_user = FactoryBot.create(:user, css_id: "HUOT_1", full_name: "Lendvai Huot", roles: ["Hearing Prep"])
hearing_day = FactoryBot.create(:hearing_day, judge: hearing_user, scheduled_for: 1.month.from_now, created_by: User.first, updated_by: User.first)
hearing = FactoryBot.create(:hearing, appeal: appeal, hearing_day: hearing_day, disposition: nil, adding_user: User.first, judge: hearing_user)
root_task = FactoryBot.create(:root_task, appeal: appeal)
hearing_task = FactoryBot.create(:hearing_task, parent: root_task, appeal: appeal, assigned_to: HearingsManagement.singleton, assigned_by: hearing_user)
association = FactoryBot.create(:hearing_task_association, hearing: hearing, hearing_task: hearing_task)
appeal.reload.treee
```

3. Confirm you cannot view the judge for a hearing, `localhost:3000/queue/appeals/${appeal.uuid}`
4. Switch to the user `TEAM_ADMIN at 101`.
5. Confirm you can still see the judge name in the hearings section.

### User Facing Changes

View for a non-VSO user (order of judge and date has changed - I pinged design to ask if I should change it back) 
BEFORE
<img width="1792" alt="Master_NonVSOCanSeeJudge_MultipleHearings" src="https://user-images.githubusercontent.com/80347702/130120117-dee7d65c-1d98-4b5e-8880-8fad503ce0b9.png"> 
AFTER
<img width="1792" alt="NonVSOCanSeeJudge_MultipleHearings" src="https://user-images.githubusercontent.com/80347702/130120206-739aeb47-4fc5-4a5e-8741-4843171dcf72.png">


View for a VSO user (it changed!)
BEFORE
<img width="1792" alt="Master_VSOCanSeeJudge_MultipleHearings" src="https://user-images.githubusercontent.com/80347702/130121418-4804a597-217e-4b0a-8c3e-e0173010e41c.png">


AFTER
<img width="1792" alt="VSOCantSeeJudge_MultipleHearings" src="https://user-images.githubusercontent.com/80347702/130121495-059976eb-c5cf-4f61-98f4-714d63ab08bd.png">



### Code Documentation Updates
